From d31718999797ffc4dd177d14760ef7b8fd95ac2f Mon Sep 17 00:00:00 2001
From: Fabio Berton <fabio.berton@ossystems.com.br>
Date: Wed, 12 Jun 2019 14:15:57 -0300
Subject: [PATCH 2/3] meson.build: make TLS GLX optional again
Organization: O.S. Systems Software LTDA.

This was optional with autotools, and needs to be disabled
when using musl C library, for instance.

Upstream-Status: Pending

Signed-off-by: Alexander Kanavin <alex.kanavin@gmail.com>
Signed-off-by: Fabio Berton <fabio.berton@ossystems.com.br>
---
 meson.build       | 4 +++-
 meson_options.txt | 7 +++++++
 2 files changed, 10 insertions(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index e11880f2902..7a07974ba61 100644
--- a/meson.build
+++ b/meson.build
@@ -362,7 +362,9 @@ if with_egl and not (with_platform_drm or with_platform_surfaceless)
   endif
 endif
 
-pre_args += '-DGLX_USE_TLS'
+if get_option('glx-tls')
+  pre_args += '-DGLX_USE_TLS'
+endif
 if with_glx != 'disabled'
   if not (with_platform_x11 and with_any_opengl)
     error('Cannot build GLX support without X11 platform support and at least one OpenGL API')
diff --git a/meson_options.txt b/meson_options.txt
index 1f72faabee8..fcd49efea27 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -339,6 +339,13 @@ option(
   value : true,
   description : 'Enable direct rendering in GLX and EGL for DRI',
 )
+option(
+  'glx-tls',
+  type : 'boolean',
+  value : true,
+  description : 'Enable TLS support in GLX',
+)
+
 option(
   'I-love-half-baked-turnips',
   type : 'boolean',
-- 
2.22.0

